pkg/tpath/tpath.go

```diff
@@ -86,7 +85,7 @@ func getPathContext(nc *PathContext, fullPath, remainPath util.Path, createMissi
 	ncNode := v.Interface()
 
 	// Hack for translation of gateway node
-	if strings.HasSuffix(pe, "gressGateway") {
+	if isGatewaysPath(pe) {
```

Comment:
Sorry for the churn - just reading over this code again I'm not sure I fully understand why this is needed here vs. when we set the tree. Because input tree for gateways does not have list elements and we should be able to determine without any mutation if ingress/egress gateways have been defined. At least that's how I see it, I may be missing something here.

Reply:
talked offline with @ostromart, the new revision would comes  implementation to have special translation to handle gateways specifically. But from some sense I still prefer the existing one because it is more natural to do it during tree traversal